asf_heat_pump_suitability/pipeline/transform/heat_network_zones.py

```
@@ -54,3 +57,64 @@ def label_gdf_heat_network_zone_uprns(
     labelled_uprn_df = pl.from_pandas(labelled_uprn_gdf.drop(columns="geometry"))
 
     return labelled_uprn_df
+
+
+def label_gdf_heat_network_spatial_signatures_uprns(
+    uprn_gdf: gpd.GeoDataFrame,
+    spatial_signatures_gdf: gpd.GeoDataFrame,
+    usecols: list = None,
+    types: list = None,
+) -> pl.DataFrame:
+    """
+    Labels UPRNs that are located within a given Spatial Signature type deemed to be potentially
+    suitable for a heat network.
+
+    Args:
+        uprn_gdf (gpd.GeoDataFrame): UPRNs with point geometries to be labelled
+        spatial_signatures_gdf (gpd.GeoDataFrame): polygons of spatial signatures
+        usecols (list, optional): names of descriptive columns from spatial_signatures_gdf to be joined with uprn_gdf (in addition to "geometry" and "type").
+            Default is None to use only "geometry" and "type".
+        types (list, optional): spatial signature types potentially suitable for a heat network. Default is None to use all spatial signature types.
```

I would probably set the default spatial signature types to those you've selected. You could put them as a global variable at the top of this file. Otherwise, here I think we could get another case of silent fail where everything gets labelled as potentially suitable for a HNZ and we can't immediately see why.